core/domain/exp_jobs_one_off.py

```diff
@@ -720,3 +720,44 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class TranslatorToVoiceArtistOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for migrating translator_ids to voice_artist_ids."""
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationRightsModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        translator_ids = item.translator_ids
+        if len(translator_ids) > 0:
+            item.voice_artist_ids = translator_ids
+            item.translator_ids = []
+            commit_message = 'Migrate from translator to voice artist'
+            commit_cmds = [{
+                'cmd': 'change_role',
+                'assignee_ids': translator_ids,
+                'old_role': 'translator',
+                'new_role': 'voice artist'
+            }]
+            item.commit('Admin', commit_message, commit_cmds)
+
+            exp_summary_model = exp_models.ExpSummaryModel.get_by_id(item.id)
+            if exp_summary_model is None or exp_summary_model.deleted:
+                yield ('Summary model does not exist or is deleted', item.id)
+            else:
+                exp_summary_model.voice_artist_ids = translator_ids
```

Comment:
@sophiewu6, Assume this one off job fails between line 748-750, and we restarted the job, in that case, will this line work correctly?

Reply:
I think this will lead to an incorrect voice_artist_ids if we run the re-job the job in any case!

Reply:
Do you mean since translator_ids was cleared at the first time, when running re-job, the translator_ids will be empty, so voice_artist_ids won't get correct values?

Reply:
since translator_ids was cleared for some models in the first incomplete run*

Yeah!

Reply:
Done.